sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -1643,4 +1643,12 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
       }
     }
   }
+
+  test("SPARK-37702: cache table with temporary function ") {
+    spark.udf.register("udf", (id: Int) => id + 1)
```

Comment:
let's wrap with `withUserDefinedFunction`, so that we can unregister the function at the end.